cli/azd/internal/appdetect/docker.go

```diff
@@ -1,19 +1,67 @@
 package appdetect
 
 import (
+	"bufio"
 	"io/fs"
+	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
 )
 
 func detectDocker(path string, entries []fs.DirEntry) (*Docker, error) {
 	for _, entry := range entries {
 		if strings.ToLower(entry.Name()) == "dockerfile" {
+			dockerFilePath := filepath.Join(path, entry.Name())
+			file, err := os.Open(dockerFilePath)
+			if err != nil {
+				return nil, err
+			}
+			defer file.Close()
+			scanner := bufio.NewScanner(file)
+
+			var ports []Port
+			for scanner.Scan() {
+				line := scanner.Text()
+				if strings.HasPrefix(line, "EXPOSE") {
+					parsedPorts, err := parsePorts(line[len("EXPOSE"):])
+					if err != nil {
+						return nil, err
+					}
+					ports = append(ports, parsedPorts...)
+				}
+			}
+
 			return &Docker{
-				Path: filepath.Join(path, entry.Name()),
+				Path:  dockerFilePath,
+				Ports: ports,
 			}, nil
 		}
 	}
 
 	return nil, nil
 }
+
+func parsePorts(s string) ([]Port, error) {
+	s = strings.TrimSpace(s)
+	var ports []Port
+	portSpecs := strings.Split(s, " ")
+	for _, portSpec := range portSpecs {
+		if strings.Contains(portSpec, "/") {
+			parts := strings.Split(portSpec, "/")
+			portNumber, err := strconv.Atoi(parts[0])
+			if err != nil {
+				return nil, err
```

Comment:
Add some context to this error:
```suggestion
				return nil, fmt.Errorf("parsing port number: %w", err)
```

Reply:
Updated.